### PR TITLE
feat: add `exp` field to JWT if `expires_in` is set

### PR DESCRIPTION
**Pull request by ericnorris**
This commit adds the ability to add an `exp` field to the signed JWT. While Terraform has some time-related functions (e.g. `timestamp` and `formatdate`), there does not appear to be a Terraform-native way of generating a _unix_ timestamp for the current time. Without this feature it would be impossible for a user to set `exp` to a meaningful value themselves.

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
google_service_account_jwt: added `expires_in` attribute for generating `exp` claim.
```

### CODE REVIEW

**Comment by ericnorris** [comment]
I don't believe the VCR test failures are related to my change, so I've marked this ready for review.

**Comment by c2thorn** [comment]
Thanks for the contribution @ericnorris! Just stating here that this is on my TODO, just have been running a bit behind on reviews. I'm not as familiar with the resource so I need to gain context first. Thanks for your patience.

**Review by c2thorn — COMMENTED** [review]
Hi @ericnorris, apologies for the delay. I spoke with my team, and there are concerns about this field playing well with Terraform's plan time validation of datasources. TF assumes that old Datasource values are valid at plan time. This field would add volatility that leave us unsure with the edge cases of.

It would be more comfortable if this was modeled as part of a resource instead, as it has more flexibility. The resource would be extremely similar in structure, but this is fairly common for other resources/datasource pairs.

**Comment by ericnorris** [comment]
Hey @c2thorn, no worries. I'm not sure this would make sense as a resource, as it would need to get destroyed and recreated on essentially every run, or else its output would be invalid.

That is, if the generated JWT _resource_ had an `expires_in` set to 60 seconds, it would create a resource who's output was only valid for 60 seconds, and you'd need to do something to taint the resource in order for it to make a valid one later. This wouldn't work for the use case I'd like to use it for, because I want it to always output a valid JWT.

I'll add that I don't think this would cause a problem with terraform at plan time, and in fact there are some time-based data sources in the provider already, see: https://github.com/GoogleCloudPlatform/magic-modules/blob/ac628318ef2e9fd4ecad1cf1916cf2ebd8d02842/mmv1/third_party/terraform/data_sources/data_source_storage_object_signed_url.go#L116

The data source generates a signed URL that will expire in some amount of time, just like my proposed change to the JWT data source, making it equally as "unstable".

**Comment by c2thorn** [comment]
> I'll add that I don't think this would cause a problem with terraform at plan time, and in fact there are some time-based data sources in the provider already, see:
> 
> https://github.com/GoogleCloudPlatform/magic-modules/blob/ac628318ef2e9fd4ecad1cf1916cf2ebd8d02842/mmv1/third_party/terraform/data_sources/data_source_storage_object_signed_url.go#L116
> 
> The data source generates a signed URL that will expire in some amount of time, just like my proposed change to the JWT data source, making it equally as "unstable".

Thanks for pointing this out, I was not aware of this. That is very similar to this case and is working. I'll move forward with this PR

File: mmv1/third_party/terraform/tests/data_source_google_service_account_jwt_test.go [hunk 1]
```diff
@@ -89,6 +99,13 @@ func TestAccDataSourceGoogleServiceAccountJwt(t *testing.T) {
 	serviceAccount := getTestServiceAccountFromEnv(t)
 	targetServiceAccountEmail := BootstrapServiceAccount(t, getTestProjectFromEnv(), serviceAccount)
 
+	staticTime := time.Now()
+
+	// Override the current time with one that will never change, to compare against later.
+	dataSourceGoogleServiceAccountJwtNow = func() time.Time {
+		return staticTime
+	}
+
```
Comment (c2thorn) [inline mmv1/third_party/terraform/tests/data_source_google_service_account_jwt_test.go:108]: I think testing a scenario where the `expires_in` is updated would help me feel like we aren't missing some edge case. Would you mind adding another step in this existing test?

**Review by c2thorn — APPROVED** [review]
updated test passed. Thank you for your effort and patience here @ericnorris

**Pull request merged**